frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/simulated-region-modify.ts

```diff
@@ -0,0 +1,69 @@
+import type { Feature } from 'ol';
```

Comment:
This whole file should be unnecessary as you could just use `viewport-modify.ts`

Reply:
For now it would be possible to use `ViewportModify`, but this is only made for `LineString` features. We already ran into the issue that we cannot specify a background color, so we want to switch to a `Polygon` feature later (see #603).

At the moment, we're not sure what consequences this change might have. So I'd either include the switch to `Polygon` in this PR (which would take some time and there are some features waiting for this PR to get finished) or keep the two implementations and do the refactoring when we're adding support for `Polygon`s, which requires changes to multiple existing implementations anyway.

Reply:
You reuse the `ViewportModify` with `LineStrings` in this PR and refactor this Modify later to a `Polygon` (the viewports would also use a Polygon  - but without a `fill`-style).

Also YAGNI.

Reply:
I've merged the two files into one